SparkCatalogMetaStoreClient.setMetaConf throws UnsupportedOperationException, blocking all Hudi Hive sync that uses it

## Bug Description

**What happened:**

`SparkCatalogMetaStoreClient` (added in #18203) throws `UnsupportedOperationException` from `setMetaConf`. `HoodieHiveSyncClient` calls `IMetaStoreClient.setMetaConf` unconditionally at construction time (to forward `hive.metastore.callerContext.*` properties for audit/tracing), so any attempt to use `SparkCatalogMetaStoreClient` via `HoodieHiveSyncClient` fails before any catalog operation can run.

This blocks the entire reason `SparkCatalogMetaStoreClient` was introduced — to avoid the Hive-on-Spark classloader split during Hive sync.

**What you expected:**

Setting `hoodie.datasource.hive_sync.use_spark_catalog=true` should make Hive sync use the Spark catalog. Instead it crashes immediately with `UnsupportedOperationException`.

**Steps to reproduce:**

1. Configure `hoodie.datasource.hive_sync.use_spark_catalog=true` (or `spark.hoodie.datasource.hive_sync.use_spark_catalog=true` once #18650 lands).
2. Write a Hudi DataFrame with `HIVE_SYNC_ENABLED=true` and `HIVE_SYNC_MODE=HIVEQL` (or HMS).
3. `HoodieHiveSyncClient` constructor invokes `setMetaConf` → `UnsupportedOperationException`.

**Root cause:**

`HoodieHiveSyncClient.setMetaConf` (HoodieHiveSyncClient.java:759-767):
```java
private void setMetaConf(HiveConf configuration) throws TException {
  Properties confProperties = configuration.getAllProperties();
  Set<String> confPropertyNames = confProperties.stringPropertyNames();
  for (String propertyName : confPropertyNames) {
    if (propertyName.startsWith("hive.metastore.callerContext")) {
      this.client.setMetaConf(propertyName, confProperties.getProperty(propertyName));
    }
  }
}
```

Called from the constructor at line 119 immediately after `createMetaStoreClient`. The properties forwarded are `hive.metastore.callerContext.*` — diagnostic metadata sent to the remote thrift HMS. With `SparkCatalogMetaStoreClient` there is no remote HMS to receive them; the values are inert.

`SparkCatalogMetaStoreClient.scala:155`:
```scala
override def setMetaConf(arg0: String, arg1: String): Unit = unsupported[Unit]()
```

A fix is in PR #<PR_NUMBER>.

## Environment

**Hudi version:** master (verified on `apache/master` as of `8d348cc6b927`)
**Query engine:** Spark (any 3.x)
**Relevant configs:** `hoodie.datasource.hive_sync.use_spark_catalog=true` with `HIVE_SYNC_ENABLED=true`

## Logs and Stack Trace

```
Caused by: java.lang.UnsupportedOperationException: Method is not supported in SparkCatalogMetaStoreClient
    at org.apache.spark.sql.hive.SparkCatalogMetaStoreClient.unsupported(SparkCatalogMetaStoreClient.scala:147)
    at org.apache.spark.sql.hive.SparkCatalogMetaStoreClient.setMetaConf(SparkCatalogMetaStoreClient.scala:155)
    at org.apache.hudi.hive.HoodieHiveSyncClient.setMetaConf(HoodieHiveSyncClient.java:764)
    at org.apache.hudi.hive.HoodieHiveSyncClient.<init>(HoodieHiveSyncClient.java:119)
    at org.apache.hudi.hive.HiveSyncTool.initSyncClient(HiveSyncTool.java:...)
    ...
```

--- a/hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/hive/SparkCatalogMetaStoreClient.scala
+++ b/hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/spark/sql/hive/SparkCatalogMetaStoreClient.scala
@@ -152,7 +152,11 @@ class SparkCatalogMetaStoreClient(syncConfig: HiveSyncConfig)
   override def isLocalMetaStore(): Boolean = unsupported[Boolean]()
   override def reconnect(): Unit = unsupported[Unit]()
   override def close(): Unit = unsupported[Unit]()
-  override def setMetaConf(arg0: String, arg1: String): Unit = unsupported[Unit]()
+  // setMetaConf is no-op: HoodieHiveSyncClient.setMetaConf forwards
+  // hive.metastore.callerContext.* values to the metastore for audit/tracing. With Spark's
+  // external catalog there is no remote HMS to forward to, so accept the call silently
+  // instead of breaking sync clients that exercise the standard IMetaStoreClient contract.
+  override def setMetaConf(arg0: String, arg1: String): Unit = {}
   override def getMetaConf(arg0: String): String = unsupported[String]()
   override def getDatabases(arg0: String): java.util.List[String] = unsupported[java.util.List[String]]()
   override def getAllDatabases(): java.util.List[String] = unsupported[java.util.List[String]]()

